packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -664,6 +664,56 @@ export class KubernetesClient {
     }
   }
 
+  async restartPod(name: string, timeoutMs: number = 60000): Promise<void> {
+    let telemetryOptions = {};
+    try {
+      const ns = this.currentNamespace;
+      const connected = await this.checkConnection();
+      if (ns && connected) {
+        const pod = await this.readNamespacedPod(name, ns);
+        if (pod?.metadata) {
+          const coreApi = this.kubeConfig.makeApiClient(CoreV1Api);
+          // To restart pod we need to delete existed pod and create the new one
+          await coreApi.deleteNamespacedPod(name, ns);
+
+          pod.metadata.resourceVersion = undefined;
+          pod.metadata.creationTimestamp = new Date();
+
+          // Wait the pod to be successfully deleted
+          await this.waitForPodDeletion(coreApi, name, ns, timeoutMs);
+
+          await coreApi.createNamespacedPod(ns, pod);
+        }
+      }
+    } catch (error) {
+      telemetryOptions = { error: error };
+      throw this.wrapK8sClientError(error);
+    } finally {
+      this.telemetry.track('kubernetesRestartPod', telemetryOptions);
+    }
+  }
+
+  private async waitForPodDeletion(coreApi: CoreV1Api, name: string, ns: string, timeoutMs: number) {
+    try {
+      await coreApi.readNamespacedPod(name, ns);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      if (error.response.statusCode === 404) {
+        return; // Pod deleted
```

Comment:
Here, you may want to use a DELETE instead of a GET, so the api won't have to compute the data to return. Also, you could use a Precondition with the uid of the pod, so you are sure to not consider another pod with the same name created in the meantime by someone else. 

Here is described how `kubectl delete` has been implemented: https://github.com/kubernetes/kubernetes/issues/42594

Reply:
Unfortunately from the code we can not use `--wait-for-deletion` option, so I tried to workaround with a bit different approach, in the code the only pod status is checking which also avoids the computation of the pod data. Also added additional checking to avoid creating the pod with the same name which is restarting at the moment.